File: Two Pointers/2. Two Sum II Input Array Is Sorted/solution.scala
object Solution {
    def twoSum(numbers: Array[Int], target: Int): Array[Int] = {
        var left = 0
        var right = numbers.length - 1
        while(left < right) {
            if (numbers(left) + numbers(right) > target) {
                right -= 1
            } else if (numbers(left) + numbers(right) < target) {
                left += 1
            } else {
                return Array(left + 1, right + 1)
            }
        }
        return Array(left, right)
    }
}

File: Two Pointers/2. Two Sum II Input Array Is Sorted/solution.rb
# @param {Integer[]} numbers
# @param {Integer} target
# @return {Integer[]}
def two_sum(numbers, target)
  left = 0
  right = numbers.length - 1
  while left < right
    if numbers[left] + numbers[right] > target
      right -= 1
    elsif numbers[left] + numbers[right] < target
      left += 1
    else
      return [left + 1, right + 1]
    end
  end
end
